ext/kernelparser/register.go

```diff
@@ -20,7 +20,12 @@ type ParseMatch struct {
 }
 
 // Parser is a kernel parser.
-type Parser func(db database.Datastore, kernelVersion, osImage string) (*ParseMatch, bool, error)
+// osImage is expected to be lowercase (for example by calling `strings.ToLower(osImage)`).
+// Returns a non-nil *ParseMatch, match, if the kernel is supported.
+// Returns ok = true if the kernel was recognized by the parser.
+// Returns a non-nil error when there is an error parsing the kernel or if the node represented by the
+// osImage and kernel is recognized and completely unsupported.
+type Parser func(db database.Datastore, kernelVersion, osImage string) (match *ParseMatch, ok bool, err error)
```

Comment:
This comment points out that `Parse` return value is getting difficult. Can we simplify the different combinations of return values and encode the different match scenarios in the match object itself? Then we can use `(match *ParseMatch, err error)` to mean:

- `match == nil && err == nil` - The parser did not recognize this os/version, but it might be recognized by another parser (same as current `ok = false`).
- `match == nil && err != nil` - The parser recognized the os/version string, but it is malformed. `Parse()` could not extract meaningful information, hence `match` is `nil`., but no other parser will recognize this os/version.
- `match != nil && err == nil` - The parser recognized and successfully parsed the os/version. 

Then `match` can have `match.isSupported` to signal the os/version is not supported. Or maybe others, so we don't have to use `ok`.

Reply:
Yeah this is getting complicated. I'm going to make specialized errors we will handle in a `switch`

Reply:
OK, will follow up there.